AK/Error.cpp

```diff
@@ -13,4 +17,26 @@ Error Error::from_string_view_or_print_error_and_return_errno(StringView string_
     return Error::from_string_view(string_literal);
 }
 
+#if defined(AK_OS_WINDOWS)
+Error Error::from_windows_error(DWORD code)
+{
+    char* message = nullptr;
+
+    auto size = FormatMessageA(
+        FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+        nullptr,
+        code,
+        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+        (LPSTR)&message,
+        0,
+        nullptr);
+
+    if (size == 0)
+        return Error::from_string_view_or_print_error_and_return_errno("Unknown error"sv, code);
+    auto error = Error::from_string_view({ message, size });
+    LocalFree(message);
+    return error;
```

Comment:
This is explicitly a use after free. StringView does not own its data. Is there not a static storage version of this API, either in Win32, or the MSVC STL?

Reply:
No, there is no static storage version as far as I know.